docs/articles/documentation/using-testcafe/configuration-file.md

```diff
@@ -0,0 +1,715 @@
+
+```json
+{
+    "filter": {
+        "fixtureMeta": {
+            "device": "mobile",
+            "env": "production"
+        }
+    }
+}
+```
+
+This configuration runs tests whose fixture's metadata has the `device` property set to `mobile` and the `env` property set to the `production`.
+
+*CLI*: [--fixture-meta](command-line-interface.md#--fixture-meta-keyvaluekey2value2)  
+*API*: [runner.filter](programming-interface/runner.md#filter)
+
+## appCommand
+
+Executes the specified shell command before running tests.
+
+```json
+{
+    "appCommand": "node server.js"
+}
+```
+
+Use the `appCommand` property to launch the application you are going to test. This application is automatically terminated after testing is finished.
+
+The [appInitDelay](#appinitdelay) property specifies the amount of time allowed for this command to initialize the tested application.
+
+> TestCafe adds `node_modules/.bin` to `PATH` so that you can use the binaries the locally installed dependencies provide without prefixes.
+
+*CLI*: [-a, --app](command-line-interface.md#-a-command---app-command)  
+*API*: [runner.startApp](programming-interface/runner.md#startapp)
+
+## concurrency
+
+Specifies the number of browser instances that should run tests concurrently.
+
+```json
+{
+    "concurrency": 3
+}
+```
+
+TestCafe opens several instances of the same browser and creates a pool of browser instances. Tests are run concurrently against this pool, that is, each test is run in the first free instance.
+
+See [Concurrent Test Execution](common-concepts/concurrent-test-execution.md) for more information about concurrent test execution.
+
+*CLI*: [-c, --concurrency](command-line-interface.md#-c-n---concurrency-n)  
+*API*: [runner.concurrency](programming-interface/runner.md#concurrency)
+
+## debugOnFail
+
+Specifies whether to automatically enter the debug mode when a test fails.
+
+```json
+{
+    "debugOnFail": true
+}
+```
+
+If this option is enabled, TestCafe pauses the test when it fails. This allows you to view the tested page and determine the cause of the fail.
+
+When you are done, click the **Finish** button in the footer to end test execution.
+
+*CLI*: [--debug-on-fail](command-line-interface.md#--debug-on-fail)  
+*API*: [runner.run({ debugOnFail })](programming-interface/runner.md#run)
+
+## appInitDelay
+
+Specifies the time (in milliseconds) allowed for an application launched using the [appCommand](#appcommand) option to initialize.
+
+TestCafe waits for the specified time before it starts running tests.
+
+```json
+{
+    "appCommand": "node server.js",
+    "appInitDelay": 3000
+}
+```
+
+**Default value**: `1000`
+
+*CLI*: [--app-init-delay](command-line-interface.md#--app-init-delay-ms)  
+*API*: [runner.startApp](programming-interface/runner.md#startapp)
+
+## selectorTimeout
```

Comment:
Maybe better to divide options into some meaningful sections

Reply:
Maybe we'll make it in a separate PR (since we'll also have to do this for CLI and it's not so obvious how to group them)?

Reply:
ok